src/main.cpp

```diff
@@ -2451,20 +2498,14 @@ void static UpdateTip(CBlockIndex *pindexNew) {
     static bool fWarned = false;
     if (!IsInitialBlockDownload() && !fWarned)
     {
-        int nUpgraded = 0;
         const CBlockIndex* pindex = chainActive.Tip();
-        for (int i = 0; i < 100 && pindex != NULL; i++)
-        {
-            if (pindex->nVersion > CBlock::CURRENT_VERSION)
-                ++nUpgraded;
-            pindex = pindex->pprev;
-        }
-        if (nUpgraded > 0)
-            LogPrintf("%s: %d of last 100 blocks above version %d\n", __func__, nUpgraded, (int)CBlock::CURRENT_VERSION);
-        if (nUpgraded > 100/2)
-        {
+        if (warningcheck.IsLockedIn(pindex, chainParams.GetConsensus())) {
             // strMiscWarning is read by GetWarnings(), called by Qt and the JSON-RPC code to warn the user:
-            strMiscWarning = _("Warning: This version is obsolete; upgrade required!");
+            if (warningcheck.IsActiveFor(pindex, chainParams.GetConsensus())) {
+                strMiscWarning = _("Warning: unknown softfork has activated");
```

Comment:
Is it really worth it to notify this twice (once when is locked in and 2016 blocks later when it's activated)?